org-formation/075-security-hub/security-hub-suppress-infra.yaml

```diff
@@ -234,22 +231,134 @@ Resources:
+                  }
+                  self.finding_identifiers.append(finding_identifier)
+
+              def get_finding_identifiers(self):
+                  return self.finding_identifiers
+
+          """
+          Uses the Lambda Powertools BatchProcessor in order to build the findings id list.
+          Lambda powertools is used to prevent successfully processed messages being returned to SQS
+          https://awslabs.github.io/aws-lambda-powertools-python/latest/utilities/batch/
+          """
+          @logger.inject_lambda_context(log_event=True)
+          def handler(event, context):
+              records = event["Records"]
+              logger.info(f"records {records}")
+
+              processor = BatchProcessor(event_type=EventType.SQS)
+
+              rh = RecordsHandler()
+              with processor(records, rh.record_handler) as proc:
+                  proc.process()
+
+              finding_identifiers = rh.get_finding_identifiers()
+              logger.info(f"finding_identifiers {finding_identifiers}")
+
+              try:
+                  response = sh_client.batch_update_findings(
+                      FindingIdentifiers=finding_identifiers,
+                      Workflow={"Status": "SUPPRESSED"},
+                  )
+                  for processed_findings in response["ProcessedFindings"]:
+                      logger.info(
+                          f"processed and suppressed id {processed_findings['Id']} productarn {processed_findings['ProductArn']}"
+                      )
+
+                  for unprocessed_findings in response["UnprocessedFindings"]:
+                      logger.error(
+                          f"unprocessed finding id {unprocessed_findings['FindingIdentifier']['Id']} productarn {unprocessed_findings['FindingIdentifier']['ProductArn']} error code {unprocessed_findings['ErrorCode']} error message {unprocessed_findings['ErrorMessage']}"
+                      )
+
+              except boto3exceptions.ClientError as error:
+                  logger.exception("client error")
+                  raise ConnectionError(f"Client error invoking batch update findings {error}")
+              except boto3exceptions.ParamValidationError as error:
+                  logger.exception("invalid parameters")
+                  raise ValueError(f"The parameters you provided are incorrect: {error}")
+
+              return {"statusCode": 200}
+      Role: !GetAtt SecurityHubBatchUpdateSuppressionRole.Arn
+      FunctionName: suppress-security-hub-findings
+      Description: Suppression lambda for Security Hub implements batch update findings
+      Handler: index.handler
+      Layers:
+      - Fn::GetAtt:
+        - AWSLambdaPowertoolsApplication
+        - Outputs.LayerVersionArn
+      Runtime: python3.8
+      Timeout: 120
+  BatchupdateEventInvokeConfig:
+    Type: AWS::Lambda::EventInvokeConfig
+    Properties:
+      FunctionName: suppress-security-hub-findings
```

Comment:
less places with hard coded names would be better therefore i would consider doing a `!Ref BatchupdateLambdaFct` here

Reply:
If I understand correctly, !Ref BatchupdateLambdaFct will return the resource name (not the function name). To make this work I can name the function the same as the resource (at line 323), then use use a Ref to the resource everywhere else. Not very elegant but should work. I've seen examples in the doc where they use !Ref for the FunctionName here, I don't understand how it works based on the doc for !Ref and AWS::Lambda::Function resource...

Reply:
i thought the function name is the same as the resource name?  if not then what would the function name look like compared to the resource name?

Reply:
When I created the lambda without a function name it generated a name starting with the resource name but ending with some (semi-)random string (same as it does for other resources where you don't specify the name). In this case it's an issue because I can't get the name as an attribute of the resource (from the doc at least), and I need it to put in the policy for the lambda. Without something like 'myFct.FctName', I don't see how I can do it without hardcoding the function name (and making it the same as the resource name is just a hack so I can use !Ref)...